client/src/components/SolutionViewer/SolutionViewer.tsx

```diff
@@ -1,66 +1,47 @@
 import { Panel } from '@freecodecamp/react-bootstrap';
 import Prism from 'prismjs';
 import React from 'react';
-import { ChallengeFile, ChallengeFiles } from '../../redux/prop-types';
+import { ChallengeFile } from '../../redux/prop-types';
 
-type SolutionViewerProps = {
-  challengeFiles: ChallengeFiles;
+type Props = {
+  challengeFiles: Solution[] | null;
   solution?: string;
 };
+type Solution = Pick<ChallengeFile, 'ext' | 'contents' | 'fileKey'>;
+
+function SolutionViewer({ challengeFiles, solution }: Props) {
+  const isLegacy = !challengeFiles || !challengeFiles.length;
+  const solutions = isLegacy
+    ? [
+        {
+          ext: 'js',
+          contents:
+            solution ?? '// The solution is not available for this project',
+          fileKey: 'script.js'
+        }
+      ]
+    : challengeFiles;
 
-function SolutionViewer({
-  challengeFiles,
-  solution = '// The solution is not available for this project'
-}: SolutionViewerProps): JSX.Element {
   return (
     <>
-      {challengeFiles?.length ? (
-        challengeFiles.map((challengeFile: ChallengeFile) => (
-          <Panel
-            bsStyle='primary'
-            className='solution-viewer'
-            key={challengeFile}
-          >
-            <Panel.Heading>{challengeFile.ext.toUpperCase()}</Panel.Heading>
-
-            <Panel.Body>
-              <pre>
-                <code
-                  dangerouslySetInnerHTML={{
-                    __html: Prism.highlight(
-                      challengeFile.contents.trim(),
-                      Prism.languages[challengeFile.ext],
-                      ''
-                    )
-                  }}
-                />
-              </pre>
-            </Panel.Body>
-          </Panel>
-        ))
-      ) : (
-        <Panel
-          bsStyle='primary'
-          className='solution-viewer'
-          key={solution.slice(0, 10)}
-        >
-          <Panel.Heading>JS</Panel.Heading>
+      {solutions.map(({ fileKey, ext, contents }) => (
+        <Panel bsStyle='primary' className='solution-viewer' key={fileKey}>
```

Comment:
```suggestion
      {solutions.map(({ ext, contents }, i) => (
        <Panel bsStyle='primary' className='solution-viewer' key={i}>
```

Reply:
Isn't it bad form to use the index of an array as the key?

Reply:
I have heard that, but,  until someone explains why, I am going to keep doing it to things I am not popping.

Reply:
It's explained here: https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

TL;DR: React will assume that components with the same keys are the same components. This can be a problem if you re-order components in a list.

We're not re-ordering them, but it's a good practice in case someone decides to.

Anyways, the issue was that `fileKey` was missing, rather than not unique.  I've fixed that now, so the warnings should go away.